applications/CompressiblePotentialFlowApplication/python_scripts/potential_flow_response_function_factory.py

```diff
@@ -1,11 +1,24 @@
 
 from KratosMultiphysics.CompressiblePotentialFlowApplication import potential_flow_response
 
+try:
+    import xmc
```

Comment:
The problem is that the import of the factory is already under the clause:

https://github.com/KratosMultiphysics/Kratos/blob/8e95a20f9984988c33a98e77074f74edf1ce6d45/applications/ShapeOptimizationApplication/python_scripts/analyzers/analyzer_internal.py#L31-L34

If the potential_flow_factory has an ImportError, the above clause prevents the error from appearing and the error message you get is thus:

https://github.com/KratosMultiphysics/Kratos/blob/8e95a20f9984988c33a98e77074f74edf1ce6d45/applications/ShapeOptimizationApplication/python_scripts/analyzers/analyzer_internal.py#L149

Which is confusing because that would not be the issue. I added the try/except in the potential flow factory to be able to throw an error here with its proper message

Reply:
Just added the import of `exaqute `as well

Reply:
So i should do what i have suggested here, in the ShapeOpt as well 🤣